diesel/src/query_dsl/mod.rs

```diff
@@ -752,6 +870,44 @@ pub trait RunQueryDsl<Conn>: Sized {
     /// `Err(NotFound)` will be returned if the query affected 0 rows. You can
     /// call `.optional()` on the result of this if the command was optional to
     /// get back a `Result<Option<U>>`
+    ///
+    /// When this method is called on an insert, update, or delete statement,
+    /// it will implicitly add a `RETURNING *` to the query,
+    /// unless a returning clause was already specified.
+    ///
+    /// # Example
```

Comment:
I think a second example showing how to use `get_result` with a select query would be great. Maybe add this as first example and then add a caption like postgres specific use case, before the existing example.

Reply:
This method isn't really meant to be used with select statements normally. `.first` is intended for that case. I don't think wanting to skip `LIMIT 1` is common enough to warrant an example here.

Reply:
What about other queries known to return only one item? For example filtering by the primary key?

Reply:
You'd still want `LIMIT 1` there.